From 3c48b94c16804572fa15aed0ad768a4ff9a973c6 Mon Sep 17 00:00:00 2001
From: Meagan Lang <langmm.astro@gmail.com>
Date: Thu, 4 Feb 2021 12:14:02 -0800
Subject: [PATCH 4/4] debug flaky zpoller test

---
 src/zpoller.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/zpoller.c b/src/zpoller.c
index 276f36de..b8728701 100644
--- a/src/zpoller.c
+++ b/src/zpoller.c
@@ -306,6 +306,8 @@ zpoller_test (bool verbose)
     zsock_t *dish = zsock_new (ZMQ_PULL);
     assert (dish);
 
+    printf("HERE 1\n");
+
     //  Set up poller
     zpoller_t *poller = zpoller_new (bowl, dish, NULL);
     assert (poller);
@@ -316,6 +318,8 @@ zpoller_test (bool verbose)
 
     zstr_send (vent, "Hello, World");
 
+    printf("HERE 2\n");
+    
     //  We expect a message only on the sink
     zsock_t *which = (zsock_t *) zpoller_wait (poller, -1);
     assert (which == sink);
@@ -325,6 +329,8 @@ zpoller_test (bool verbose)
     assert (streq (message, "Hello, World"));
     zstr_free (&message);
 
+    printf("HERE 3\n");
+    
     //  Stop polling reader
     rc = zpoller_remove (poller, sink);
     assert (rc == 0);
@@ -334,6 +340,8 @@ zpoller_test (bool verbose)
     assert (rc == -1);
     assert (errno == EINVAL);
 
+    printf("HERE 4\n");
+    
     //  Check we can poll an FD
     rc = zsock_connect (bowl, "tcp://127.0.0.1:%d", port_nbr);
     assert (rc != -1);
@@ -343,6 +351,8 @@ zpoller_test (bool verbose)
     zstr_send (vent, "Hello again, world");
     assert (zpoller_wait (poller, 500) == &fd);
 
+    printf("HERE 5\n");
+    
     // Check zpoller_set_nonstop ()
     zsys_interrupted = 1;
     zpoller_wait (poller, 0);
@@ -358,6 +368,8 @@ zpoller_test (bool verbose)
     zsock_destroy (&bowl);
     zsock_destroy (&dish);
 
+    printf("HERE 6\n");
+    
 #ifdef ZMQ_SERVER
     //  Check thread safe sockets
     zpoller_destroy (&poller);
@@ -374,6 +386,8 @@ zpoller_test (bool verbose)
 
     zstr_send (client, "Hello, World");
 
+    printf("HERE 7\n");
+    
     //  We expect a message only on the server
     which = (zsock_t *) zpoller_wait (poller, -1);
     assert (which == server);
-- 
2.26.2

