--- /root/czmq-4.1.1/include/czmq_prelude.h	2018-05-15 11:09:01.000000000 -0500
+++ /root/czmq-4.1.1/include/czmq_prelude_new.h	2018-07-18 10:35:46.000000000 -0500
@@ -212,6 +212,16 @@
 #   define __UTYPE_GENERIC
 #endif
 
+//- Include stdint.h first to ensure all types initialized
+
+#if defined _MSC_VER && _MSC_VER < 1600
+/* The c99 defining the limit macros (UINT32_MAX for example), says:
+ * C++ implementations should define these macros only when __STDC_LIMIT_MACROS
+ * is defined before <stdint.h> is included. */
+#define __STDC_LIMIT_MACROS
+#include <stdint.h>
+#endif
+
 //- Always include ZeroMQ headers -------------------------------------------
 
 #include "zmq.h"
