--- /root/czmq-4.1.1/include/czmq_prelude.h	2018-05-15 11:09:01.000000000 -0500
+++ /root/czmq-4.1.1/include/czmq_prelude_new.h	2018-07-17 20:47:50.000000000 -0500
@@ -212,6 +212,12 @@
 #   define __UTYPE_GENERIC
 #endif
 
+//- Include stdint.h first to ensure all types initialized
+
+#if defined _MSC_VER && _MSC_VER < 1600
+#include "stdint.h"
+#endif
+
 //- Always include ZeroMQ headers -------------------------------------------
 
 #include "zmq.h"
